packages/ui/src/components/va-counter/VaCounter.vue

```diff
@@ -0,0 +1,391 @@
+  { listen: 'click-append-inner', emit: 'click:increase-icon' },
+])
+
+export default defineComponent({
+  name: 'VaCounter',
+
+  components: { VaInputWrapper, VaIcon, VaButton },
+
+  props: {
+    ...useFormProps,
+    ...useStatefulProps,
+    // input
+    modelValue: { type: [String, Number] as PropType<string | number>, default: 0 },
+    manualInput: { type: Boolean as PropType<boolean>, default: false },
+    stateful: { type: Boolean as PropType<boolean>, default: false },
+    min: { type: Number as PropType<number>, default: undefined },
+    max: { type: Number as PropType<number>, default: undefined },
+    step: { type: Number as PropType<number>, default: 1 },
+    label: { type: String as PropType<string>, default: '' },
+    // hint
+    messages: { type: [Array, String] as PropType<string[] | string>, default: () => [] },
+    // style
+    width: { type: [String, Number] as PropType<string | number>, default: '160px' },
+    color: { type: String as PropType<string>, default: 'primary' },
+    outline: { type: Boolean as PropType<boolean> },
+    bordered: { type: Boolean as PropType<boolean> },
+    // icons & buttons
+    increaseIcon: { type: String as PropType<string>, default: 'add' },
+    decreaseIcon: { type: String as PropType<string>, default: 'remove' },
+    buttons: { type: Boolean as PropType<boolean>, default: false },
+    flat: { type: Boolean as PropType<boolean>, default: true },
+    rounded: { type: Boolean as PropType<boolean>, default: false },
+    margins: { type: [String, Number] as PropType<string | number>, default: '4px' },
+    textColor: { type: String as PropType<string | undefined>, default: undefined },
+  },
+
+  emits: [
+    'update:modelValue',
+    ...createInputEmits(),
+    ...createFieldEmits(),
+    ...useFocusEmits,
+  ],
+
+  inheritAttrs: false,
+
+  setup (props, { emit, attrs }) {
+    const input = ref<HTMLInputElement | HTMLDivElement | undefined>()
+
+    const {
+      isFocused,
+      // will be useful when we resolve problem with 'withConfigTransport'
+      focus,
+      blur,
+    } = useFocus(input, emit)
+
+    const { valueComputed } = useStateful(props, emit)
+
+    const setCountInput = ({ target } : { target: HTMLInputElement }) => {
+      valueComputed.value = Number(target?.value)
+    }
+
+    const setCountChange = ({ target } : { target: HTMLInputElement }) => {
+      calculateCounterValue(Number(target?.value))
+    }
+
+    const getRoundDownWithStep = (value: number) => {
+      if (!props.min || !props.step) { return value }
+      return props.min + props.step * Math.floor((value - props.min) / props.step)
+    }
+
+    const calculateCounterValue = (counterValue: number) => {
+      if (props.min && counterValue < props.min) {
+        valueComputed.value = props.min
+        return
+      }
+
+      if (props.max && (counterValue > props.max)) {
+        valueComputed.value = getRoundDownWithStep(props.max)
```

Comment:
```suggestion
        valueComputed.value = props.max
```

Can't it be just props.max?

Reply:
Not always, since the `step` may not be a multiple of the `max - min`

Reply:
So if component has value 24 with step=4 and max=25 there is no way to go upper more than 24. Make sense.
Okay.